dxp/liferay_global_event_handler.md

```diff
@@ -0,0 +1,213 @@
+
+##### Code example:
+
+```js
+var onDestroyPortlet = function () {
+	Liferay.detach('messagePosted', onMessagePosted);
+	Liferay.detach('destroyPortlet', onDestroyPortlet);
+};
+
+Liferay.on('destroyPortlet', onDestroyPortlet);
+```
+
+### `Liferay.once`:
+
+Description from [yui3/event-target.js at master · liferay/yui3 · GitHub](https://github.com/liferay/yui3/blob/master/src/event-custom/js/event-target.js#L136):
+
+Listen to a custom event hosted by the global event handler one time.
+
+This is the equivalent to `on` function except the listener is immediately detached when it is executed.
+
+##### Code Example:
+
+```js
+Liferay.once(this.ns('formReady'), (event) => {
+	const form = Liferay.Form.get(event.formName);
+
+	const field = form.formValidator.getField(this.ns('emailAddress'));
+
+	if (field) {
+		if (this.viewValidDomainsURL) {
+			this.addFieldMessage_(field);
+		}
+	}
+});
+```
+
+### `Liferay.onceAfter`:
+
+Description from [yui3/event-target.js at master · liferay/yui3 · GitHub](https://github.com/liferay/yui3/blob/master/src/event-custom/js/event-target.js#L158):
+
+Listen to a custom event hosted by the global event handler one time.
+
+This is the equivalent to `after` except the listener is immediately detached when it is executed.
+
+##### Code Example:
+
+```js
+Liferay.onceAfter(
+    '<%= portletDisplay.getId() %>:messagePosted',
+    function (event) {
+        <%= randomNamespace %>onMessagePosted(
+            response,
+            refreshPage
+        );
+    }
+);
+
+```
+
+### `Liferay.after`:
+
+Description from [yui3/event-target.js at master · liferay/yui3 · GitHub](https://github.com/liferay/yui3/blob/master/src/event-custom/js/event-target.js#L848):
+
+Subscribe to a custom event hosted by the global event handler.
+
+The supplied callback will execute after any listeners add via the subscribe method, and after the default function, if configured for the event, has executed.
+
+##### Code Example:
+
+```js
+Liferay.after('commerce:productAddedToCart', function (event) {
+	Liferay.Portlet.refresh('#p_p_id<portlet:namespace />');
+});
+```
+
+### `Liferay.before`:
+
+Description from: [yui3/event-target.js at master · liferay/yui3 · GitHub](https://github.com/liferay/yui3/blob/master/src/event-custom/js/event-target.js#L886)
+
+Executes the global event callback before a DOM event, custom event or method. If the first argument is a function, it is assumed the target is a method.
+
+### `Liferay.detach`
+
+Description from: [yui3/event-target.js at master · liferay/yui3 · GitHub](https://github.com/liferay/yui3/blob/master/src/event-custom/js/event-target.js#L362)
+
+Detach one or more listeners the from a specified event
+
+##### Code Example
+
+```js
+var onDestroyPortlet = function () {
+	delegateHandler.removeListener();
+
+	Liferay.detach('destroyPortlet', onDestroyPortlet);
+};
+
+Liferay.on('destroyPortlet', onDestroyPortlet);
```

Comment:
Not sure if this is needed but I think we want to use es6 in code examples; and we prefer function declarations over function expressions - (@wincent will confirm if I'm wrong)

Reply:
Updated!